File: AutoStock.Client/src/types/notification.ts
export type NotificationType =
  1 | 2 | 3 | 4;


export interface Notification {
  id: number;

  type: NotificationType;

  title: string;

  message: string;

  entityName: string | null;

  entityId: string | null;

  isRead: boolean;

  readAt: string | null;

  createdAt: string;
}


export interface UnreadNotificationCount {
  count: number;
}

File: AutoStock.Domain/Entities/Notification.cs
using AutoStock.Domain.Enums;

namespace AutoStock.Domain.Entities;

public class Notification
{
    public int Id { get; set; }

    public string RecipientUserId { get; set; }
        = string.Empty;

    public NotificationType Type { get; set; }

    public string Title { get; set; }
        = string.Empty;

    public string Message { get; set; }
        = string.Empty;

    public string? EntityName { get; set; }

    public string? EntityId { get; set; }

    public bool IsRead { get; set; }
        = false;

    public DateTime? ReadAt { get; set; }

    public DateTime CreatedAt { get; set; }
        = DateTime.UtcNow;
}